deployer/deployer-services/src/main/java/io/hyscale/deployer/services/builder/AppMetadataBuilder.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.deployer.services.builder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.commons.utils.ResourceLabelUtil;
+import io.hyscale.deployer.core.model.AppMetadata;
+import io.kubernetes.client.openapi.models.V1Pod;
+
+@Component
+public class AppMetadataBuilder {
+
+    /**
+     * Gets app and service name from labels generated by hyscale
+     * For pods not deployed with hyscale AppMetadata with only namespace will be returned
+     * @param podList
+     * @return List of {@link AppMetadata} containing details of deployed apps
+     */
+    public List<AppMetadata> listApplications(List<V1Pod> podList) {
+        if (podList == null) {
+            return null;
+        }
+        Map<String, AppMetadata> mapping = new HashMap<String, AppMetadata>();
+
+        podList.forEach(pod -> {
+            String namespace = pod.getMetadata().getNamespace();
+            String appName = ResourceLabelUtil.getAppName(pod.getMetadata().getLabels());
+            String serviceName = ResourceLabelUtil.getServiceName(pod.getMetadata().getLabels());
+
+            if (mapping.get(namespace) == null) {
+                AppMetadata appData = new AppMetadata();
+                appData.setNamespace(namespace);
+                mapping.put(namespace, appData);
+            }
+            // One namespace can have only one app
+            if (StringUtils.isBlank(appName)) {
```

Comment:
Builder will not bother about appname / service name being blank, he will just build and give and controller will take care of the listing them to user

Reply:
This was to cover the case where label doesn't have app name but have service name, in that case we will ignore the service as well.

Reply:
if (StringUtils.isBlank(serviceName) || StringUtils.isBlank(appname) ) {
                return;
            }
Why do we have such scenarios ? It is an invalid state so we can just return I guess ? What is your point on this ?

Reply:
Such scenarios will only occur with manually applied yamls. Not in normal deployment.
If only app name is present, I think we should display the app name. in that case service name would be blank.

Reply:
Why is a service deployed without its label, does it considered to be a hyscale deployment ? But from a end user point of view for listing , if we  show  the appname but the servicename  , does it give a wrong perception to that he might think service is not deployed ?

Reply:
I agree, this might give a false information to user if no service name is shown along with app.
Even with status command we will only show services deployed through hyscale.
will update the check to 
if (StringUtils.isBlank(serviceName) || StringUtils.isBlank(appname) ) {
return;
}